docs/data-plane.md

```diff
@@ -0,0 +1,60 @@
+# Data Plane Specification
+
+This document is a work in progress to disucss the data plane requirements for machine learning serving.
+
+Its aims:
+
+ * Provide a set of schemas for generic machine learning  components, such as model servers, explainers, outlier detectors.
+ * Provide a set of proposals for components to advertise the schemas they support.
+
+## Data Plane Schemas
+
+There are various components that are useful for ML inference, these include
+
+ * The core machine learning model
+ * Model explanation
+ * Outlier detection
+ * Concept drift (skew) detection
+
+Schemas will be needed for each.
+
+### Models
+
+Data planes for request/response to machine learning models are the most well defined in the ecosystem. Existing examples are:
+
+ * [Tensorflow Serving](https://github.com/tensorflow/serving/blob/master/tensorflow_serving/apis/prediction_service.proto)
+ * [Seldon Core](https://github.com/SeldonIO/seldon-core/blob/master/proto/prediction.proto)
+
+It is suggested we don't define a new data plane for model input/output at present but allow models to publish the input/output schema they respect.
+
+
+### Model Explanation
+
+TODO
+
+### Outlier Detection
+
+TODO
+
+### Concept Drift
+
+TODO
+
+
+## Combined Schema
+
+There is an open question whether we define a combined schema to return the aggregation from the various components or we assume only the model response is returned and other components (model explanation etc) return their response asychronously to some metrics/logging channel.
+
+
+## Metadata
+
+  * There should be a unified prediction id so responses from varied components can be tied together for monitoring and auditing.
+
+It unclear whether we should impose any other metadata.
```

Comment:
I know this is WIP draft, but consider adding a note about turning on or off features (explain/skew) from the data plane